### PR TITLE
NNS1-2851: Clean transactions services and api

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Unify wallet implementation.

In this PR, simplify the transactions api and services.

Before: component used `loadWalletTransactions` that used `loadIcrcAccountTransactions` with `getTransactions` from icrc-index api which used `getTransactions` from wallet-api.

Now: component uses `loadIcrcAccountTransactions` which uses `getTransactions` from icrc-index api.

# Changes

* Remove unused sns-index.api. It was used in SnsWallet before.
* Remove wallet-index api.
* Remove wallet-transaction services.
* icrc-transaction services use directly the icrc-index api functions.
* icrc-index api uses canister directly instead of a callback.
* IcrcWalletTransactionList uses `loadIcrcAccountTransactions` and `loadIcrcAccountNextTransactons`.
* ckbtc-convert services uses `loadIcrcAccountTransactions`.

# Tests

* Add test file for icrc-transaction services.
* Change test of icrc-index api to test calls to the canister instead of the callback.
* Mock icrc-transactions instead of wallet-transactions in tests.
* Remove sns-index api spec file.
* Remove wallet-index api spec file.
* Remove wallet-transaction services spec file.

# Todos

- [ ] Add entry to changelog (if necessary).

Not necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4474** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4474?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4474?utm_source=stack-comment).

File: frontend/src/tests/lib/services/ckbtc-accounts.services.spec.ts [hunk 1]
```diff
@@ -8,12 +8,6 @@ import { resetIdentity } from "$tests/mocks/auth.store.mock";
 import { mockCkBTCMainAccount } from "$tests/mocks/ckbtc-accounts.mock";
 import { mockTokens } from "$tests/mocks/tokens.mock";
 
-vi.mock("$lib/services/wallet-transactions.services", () => {
-  return {
-    loadCkBTCAccountTransactions: vi.fn().mockResolvedValue(undefined),
```
Comment (lmuntaner) [inline frontend/src/tests/lib/services/ckbtc-accounts.services.spec.ts:13]: This didn't exist anymore.

File: frontend/src/lib/services/icrc-transactions.services.ts [hunk 2]
```diff
@@ -68,20 +50,27 @@ export const loadIcrcAccountTransactions = async ({
   }
 };
 
+export type LoadIcrcAccountNextTransactions = {
```
Comment (dskloetd) [inline frontend/src/lib/services/icrc-transactions.services.ts:53]: Should this have `Params` in the name?

File: frontend/src/lib/services/icrc-transactions.services.ts [hunk 3]
```diff
@@ -68,20 +50,27 @@ export const loadIcrcAccountTransactions = async ({
...
-  canisterId,
-  loadAccountTransactions,
+  universeId,
+  indexCanisterId,
 }: LoadIcrcAccountNextTransactions) => {
   const store = get(icrcTransactionsStore);
   const currentOldestTxId = getOldestTxIdFromStore({
     account,
-    canisterId,
+    canisterId: universeId,
```
Comment (dskloetd) [inline frontend/src/lib/services/icrc-transactions.services.ts:67]: I would expect `canisterId` to be the index canister ID and I would expect `universeId` to be the ledger canister ID.
Can you make it clearer from the names what these are?

File: frontend/src/tests/lib/services/icrc-transactions.services.spec.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        start,
+        indexCanisterId,
+        universeId: universeCanisterId,
+      });
+
+      const account = {
+        owner: mockCkBTCMainAccount.principal,
+      };
+
+      expect(spyGetTransactions).toBeCalledWith({
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/icrc-transactions.services.spec.ts:53]: Expect it to be called once?
And 0 times before the service call?

File: frontend/src/tests/lib/services/icrc-transactions.services.spec.ts [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        account: mockCkBTCMainAccount,
+        indexCanisterId,
+        universeId: universeCanisterId,
+      });
+
+      const account = {
+        owner: mockCkBTCMainAccount.principal,
+      };
+
+      expect(spyGetTransactions).toBeCalledWith({
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/icrc-transactions.services.spec.ts:146]: Expect number of calls before and after?

File: frontend/src/tests/lib/services/icrc-transactions.services.spec.ts [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/icrc-transactions.services.spec.ts:183]: same

**Review by dskloetd — COMMENTED** [review]
So good to get rid of these callbacks!

File: frontend/src/lib/services/icrc-transactions.services.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/services/icrc-transactions.services.ts]: done1

File: frontend/src/lib/services/icrc-transactions.services.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/lib/services/icrc-transactions.services.ts]: I removed the `canisterId` as parameter.

But I don't want to use ledger canister id. Instead, there are indexCanisterId and universeId.

One is used as key to store and read data. The other one to make the call.

File: frontend/src/tests/lib/services/icrc-transactions.services.spec.ts (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/icrc-transactions.services.spec.ts:55]: done

File: frontend/src/tests/lib/services/icrc-transactions.services.spec.ts (hunk 5 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/icrc-transactions.services.spec.ts:150]: done

File: frontend/src/tests/lib/services/icrc-transactions.services.spec.ts (hunk 6 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/icrc-transactions.services.spec.ts:189]: done

File: frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte [hunk 7]
```diff
@@ -51,9 +51,9 @@
 
   const loadNextTransactions = async () => {
     loading = true;
-    await loadWalletNextTransactions({
+    await loadIcrcAccountNextTransactions({
       account,
-      canisterId: ledgerCanisterId,
+      universeId: ledgerCanisterId,
```
Comment (dskloetd) [inline frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte:56]: `IcrcWalletTransactionsList` is used by `SnsWallet`:
https://github.com/dfinity/nns-dapp/blob/918f6294e9cb0232d07339bdf4f32784430f6dfc/frontend/src/lib/pages/SnsWallet.svelte#L50

`canisterId` was too vague, but `universeId` is actually wrong if the universe is an SNS universe.
If you change the name of the param, it should be named `ledgerCanisterId`.

File: frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte [hunk 8]
```diff
@@ -78,9 +78,9 @@
     // We optimistically try to fetch the new transaction the user just transferred by delaying the reload of the transactions.
     await waitForMilliseconds(WALLET_TRANSACTIONS_RELOAD_DELAY);
 
-    await loadWalletTransactions({
+    await loadIcrcAccountTransactions({
       account,
-      canisterId: ledgerCanisterId,
+      universeId: ledgerCanisterId,
```
Comment (dskloetd) [inline frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte:83]: same

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

File: frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte (hunk 7 above)
Comment (dskloetd) [reply frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte:56]: This doesn't seem to be addressed?

Comment (lmuntaner) [reply frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte]: Done

File: frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte (hunk 8 above)
Comment (lmuntaner) [reply frontend/src/lib/components/accounts/IcrcWalletTransactionsList.svelte]: Done

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Pull request merged**